skyplane/planner/planner.py

```diff
@@ -309,28 +284,46 @@ def _calculate_vm_types(self, region_tag: str) -> Optional[Tuple[int, int]]:
             return None
 
         config_vm_type = getattr(self.transfer_config, f"{cloud_provider}_instance_class")
-        config_vcpus = Planner._vm_to_vcpus(cloud_provider, config_vm_type)
+        config_vcpus = self._vm_to_vcpus(cloud_provider, config_vm_type)
         if config_vcpus <= quota_limit:
-            return config_vcpus, quota_limit // config_vcpus
+            return config_vm_type, quota_limit // config_vcpus
 
-        vcpus = None
-        for value in Planner._VCPUS:
-            if value <= quota_limit:
-                vcpus = value
+        vm_type, vcpus = None, None
+        for instance_name, vcpu_cost in sorted(self.vcpu_info[cloud_provider].items(), key=lambda x: x[1], reverse=True):
+            if vcpu_cost <= quota_limit:
+                vm_type, vcpus = instance_name, vcpu_cost
                 break
 
         # shouldn't happen, but just in case we use more complicated vm types in the future
-        if vcpus is None:
+        if vm_type is None or vcpus is None:
             return None
 
         # number of instances allowed by the quota with the selected vm type
         n_instances = quota_limit // vcpus
```

Comment:
makes sense yes. Adding this logic for all planners + accounting for the n_instances. SInce I am getting the minimum of all n_instances, I can just add self.n_instances to planner.py L158